### PR TITLE
InstancedGeometry: added a field to keep track of all instance data b…

### PR DESCRIPTION
**Pull request by Ali-RS**
…uffers (transformInstanceData and globalInstanceData) so we do not need to recreate it every frame which was not good performance-wise.

### CODE REVIEW

**Comment by Ali-RS** [comment]
Unless there's substantial discussion, I plan to integrate this in about 24 hours.

File: jme3-core/src/main/java/com/jme3/scene/instancing/InstancedGeometry.java [hunk 1]
```diff
@@ -383,14 +389,18 @@ protected void updateWorldBound() {
 
     @SuppressWarnings("unchecked")
```
Comment (stephengold) [inline jme3-core/src/main/java/com/jme3/scene/instancing/InstancedGeometry.java:390]: I think you can re-enable this warning.

File: jme3-core/src/main/java/com/jme3/scene/instancing/InstancedGeometry.java [hunk 2]
```diff
@@ -383,14 +389,18 @@ protected void updateWorldBound() {
...
     public VertexBuffer[] getAllInstanceData() {
-        ArrayList<VertexBuffer> allData = new ArrayList();
+        return allInstanceData;
+    }
+
+    private void updateAllInstanceData() {
```
Comment (stephengold) [inline jme3-core/src/main/java/com/jme3/scene/instancing/InstancedGeometry.java:395]: I think the `read()` method needs to invoke `updateAllInstanceData()` in order to ensure that `allInstanceData` gets initialized.

File: jme3-core/src/main/java/com/jme3/scene/instancing/InstancedGeometry.java [hunk 3]
```diff
@@ -66,6 +66,9 @@
     private VertexBuffer[] globalInstanceData;
     private VertexBuffer transformInstanceData;
     private Geometry[] geometries = new Geometry[1];
+    // Keep track of both transformInstanceData and globalInstanceData
+    // going to be used by renderer.
```
Comment (stephengold) [inline jme3-core/src/main/java/com/jme3/scene/instancing/InstancedGeometry.java:70]: Why "going to be"?

**Review by stephengold — CHANGES_REQUESTED** [review]

File: jme3-core/src/main/java/com/jme3/scene/instancing/InstancedGeometry.java (hunk 2 above)
Comment (Ali-RS) [reply jme3-core/src/main/java/com/jme3/scene/instancing/InstancedGeometry.java:394]: Done.

**Comment by Ali-RS** [comment]
@stephengold thanks for reviewing.

**Pull request merged**